### ISSUE TITLE
3rd Community Review of NonEntropy Allocator

### ISSUE THREAD

**Issue by joshua-ne** [filecoin-project/Allocator-Governance#228]
Allocator Application: https://github.com/filecoin-project/notary-governance/issues/1022
First Community Diligence: https://github.com/filecoin-project/Allocator-Governance/issues/63
Second Community Diligence: https://github.com/filecoin-project/Allocator-Governance/issues/148
Allocator Report: https://compliance.allocator.tech/report/f03018029/1731979795/report.md

Since the last community diligence and we got a refresh of 5 PiB Datacap, the following are the updates:
1. **New Client** 
We have served two more clients: https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/34, https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/39, and https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/41

2. **Progressive Allocation**
We continue to manage the Clients' application and approval process in accordance with the guidelines of DataCap governance. We always start with a small amount, building trust by checking data distribution and SP retrieval rate. For some of the suspicious behavior, we adjusted a little bit, for example, with https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/39, we allocated two consecutive 50TiB to begin with, since they did not update their change of SPs on time. 

3. **Data Distribution**
Our Clients allocate DataCapto different SPs in reasonable proportions and ensure that SPs do not store duplicate CIDs.

4. **Emphasis on Data Retrieval**
We have been using both SPARK and our own retrieval tool, which is now OPEN SOURCE (and we have introduced this new efficient tool to the community at a meeting, https://www.youtube.com/watch?v=XQlyGV4N_y8, starting 44:10)  and which we believe is more suitable for allocators' usage. For example, we have seen SPs taking advantage of the SPARK system by selectively broadcasting deals, say only 10% or even 1% of them, but still get a good retrieval rate. However, this method will not pass our checking, since we check on the deals made between specific clients and SPs with given DataCids of specific batches. We always check before we approve, and we have rejected several signings due to the fact that they do not perform well on either SPARK or our own retrieval system. 

5. **Dataset Card Initiative to Make Data More Valuable**
To help the community better use the data onboarded on Fil+, we have started a new initiative called Dataset Card, https://github.com/joshua-ne/FIL_DC_Allocator_1022_Dataset_Card/issues/1. We are testing and practicing on our past and ongoing clients. With this mechanism established, the community will have much better access to the real data on Filecoin which may lead to easier development on Filecoin. 

As we reach out to more clients and SPs, we understand more about their needs and challenges in onboarding data to Filecoin network. From what we've seen, the majority of them are willing to store and keep the unsealed copy of the deal data, it is just we need to do more work to make the onboarding and retrieval easier. 

As our operations expand, we are collaborating with an increasing number of clients, which has led to a faster utilization of quotas and shorter application cycles. Therefore, we hope to secure a higher quota in this round. We deeply value the principles of FIL+ and have consistently been actively involved in the FIL+ community. We have contributed tools such as retrieval platforms and CID databases, making these mature tools open-source for the community. With this commitment, we will continue to handle and utilize the granted quota responsibly and prudently.

**Comment by filecoin-watchdog** [comment]
@joshua-ne 
[Allocator Application](https://github.com/filecoin-project/notary-governance/issues/1022)  
[Compliance Report](https://compliance.allocator.tech/report/f03018029/1732757079/report.md)  
[1st Review](https://github.com/filecoin-project/Allocator-Governance/issues/63)  
[2nd Review](https://github.com/filecoin-project/Allocator-Governance/issues/148)  
1st Review score: 5PiB
2nd Review score: 10PiB

8.5 PiB granted to existing clients:
| Client Name | DC | status |
| -------- | ------- | ------- |
| US. General Services Administration | 3PiB | Existing |
| EuropeanOrganization for Nuclear Research | 868TiB | New |
| PangeoCommunity | 1.9PiB | New |
| GoogleDeepMind | 2.8PiB | New |

[US General Services Administration](https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/29)
* The client declared 5 replicas; why are there 9?
* Replication is suboptimal: 48.91% of deals are for data replicated across fewer than 4 storage providers.
* According to the Spark retrieval report, 6 SPs have retrieval rates that are too low.

[European Organization for Nuclear Research](https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/39)
* Did the allocator perform KYC?
* The allocator is responsible for updating the SP list and ensuring the client does so as well.
* The client declared 5 replicas, but 7 have already been made.
* According to the Spark retrieval report, 2 SPs have retrieval rates below 30%.

[Pangeo Community](https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/41)
* Did the allocator perform KYC?
* My research indicates that the last sealing of this data occurred in 2023, not 2021: https://github.com/filecoin-project/filecoin-plus-large-datasets/issues/1891. On what basis does the client suspect this dataset is not retrievable?
* All SPs match the provided list.

[Google DeepMind](https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/34)
* Did the allocator perform KYC?
* The client requested 10PiB but claims the dataset is 2PiB, with only 4 replicas planned, which totals 8PiB. This is inconsistent.
* All SPs match the provided list.
* The client declared 4 replicas, but 7 have already been made.
* Why 2nd allocation was 256TiB instead of 100TiB?

---

* The allocator ensures that the SP list is always up-to-date and reflects the latest information in the form.
* There is no clear evidence of KYC being performed; please clarify this matter.
* The allocator consistently monitors client retrieval using both proprietary tools and common reports. This demonstrates the allocator's strong commitment to thoroughly vet its clients and maintain high standards for the network.
* The allocator has introduced an additional solution for its clients in the form of a dataset card. Dear @joshua-ne, could you provide more information about this solution? What is its intended purpose, and where are the responses from clients who complete this form stored? I would appreciate more details on this.

**Comment by joshua-ne** [comment]
Question 1: Replica Difference
Some clients claimed 4-5 replicas, but 7-9 replicas are stored; some of the replicas are suboptimal, with fewer than 4 stored.
Response:
We initially interpreted the replication number specified in the application form as a minimum requirement, rather than a strict limit. As a result, we did not focus heavily on cases where the number of replicas exceeded the stated minimum. However, we will now reach out to the relevant clients to understand the rationale behind storing more than 4 replicas. Possible reasons could include increased data importance or coordination challenges among SPs.
Regarding cases where fewer than 4 replicas were stored, our understanding is that these discrepancies are primarily due to transmission difficulties faced by clients in transferring certain datasets to specific SPs. We are closely monitoring these instances and will encourage the clients to create additional replicas to meet the required thresholds. This matter will remain a priority to ensure compliance with the agreed replication requirements.
________________________________________
Question 2: A Few SPs Are Not Performing Well on SPARK Retrieval
Response:
We are aware that some SPs are not performing optimally on SPARK retrieval. After evaluating various factors, we developed and open-sourced our own retrieval bot to complement SPARK’s capabilities. Key considerations include:
1.  SPARK's Limitations:
•   SPARK cannot differentiate between deals made by a specific client versus other clients, which introduces noise in the retrieval performance metrics.
•   The system is sensitive to multiple variables, such as changes in multi-addresses, peerIDs, and the diverse deal-handling methods employed by SPs. For instance, some SPs may use custom optimizations instead of relying solely on Boost.
2.  Retrieval Speed and Accuracy:
•   SPARK’s data collection process is time-consuming, leading to delays in assessing retrieval performance.
•   Retrieval rates can fluctuate due to network upgrades or temporary SP settings. For instance, certain SPs have demonstrated good retrieval rates initially but underperformed following updates, while others improved retrieval metrics through questionable methods.
To address these limitations, we developed a new retrieval bot that offers:
•   Greater accuracy and immediacy in evaluating client-SP pairs.
•   A focus on specific client datasets, avoiding external influences.
•   Resistance to manipulation, ensuring reliable results.
The bot is now open-sourced and includes an API for verifying the retrieval performance of specific data CIDs for a given client-SP pair. This complements SPARK by providing a faster, more precise, and cheat-resistant solution to assess retrieval capabilities.
________________________________________
Question 3: Second Allocation Was 256 TiB Instead of 100 TiB
Response:
As outlined in the corresponding GitHub issue ((https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/34#issuecomment-2448905953)), this client demonstrated unique circumstances. Their SPs were highly "sealing-hungry," having rented sealing servers to minimize costs by completing sealing as quickly as possible. Initially, we rejected their request to sign the allocation earlier because their retrieval rate was unsatisfactory on both retrieval systems.
We granted this one-time exception of a 256 TiB allocation only after observing a significant improvement in their retrieval rate.
Additionally, we believe a degree of flexibility in batch sizes is reasonable. Most allocators follow a pattern of 100/128 TiB for the first batch and 200/256 TiB for the second. By this standard, our approach was cautious. For example, in another case ([DataCap Application] Byte Tunneling - ByteTunneling_data_store_bc_fil_01), we allocated two consecutive 50 TiB batches initially because the client failed to update their SP changes promptly.
We hope this demonstrates that our flexibility is applied judiciously to balance fairness with practicality.
________________________________________
Question 4: KYC
Response:
We acknowledge that we have yet to identify a suitable and cost-effective third-party KYC provider to partner with. However, we take this issue seriously and have made efforts to address it within our current resources.
To that end, we studied the KYC procedures adopted by other allocators and created a GitHub repository for clients to submit their details (https://github.com/joshua-ne/FIL_DC_Allocator_1022_Client_DD/issues/new?assignees=joshua-ne&labels=&projects=&template=fil_dc_allocator_1022_client_dd.yml&title=%5BFIL_DC_Allocator_1022_Client_DD%5D%3A+%3CProject+Name%3E). Unfortunately, this approach has not been as effective as we hoped, primarily because the information collected through this method largely overlaps with the content in the clients' initial application forms. As a result, we have not fully implemented this process yet.
Since most of our clients are data preparers of public datasets, we have shifted our focus to conducting more rigorous KYC checks for the storage providers (SPs) they collaborate with. The steps we routinely take include:
1. Geolocation Verification: checking if SPs are using VPNs to mask their actual geographic locations. For example, we use tools like Teoh VPN Detection(https://ip.teoh.io/vpn-detection) to verify this.
2. Cross-Referencing SP Ownership: collecting the owner, worker, and controller addresses of the SPs associated with a client and performing cross-references to ensure that these SPs are not controlled by the same entity, thereby maintaining decentralization and data security.
While these measures reflect our commitment to addressing KYC concerns, we recognize that there is room for improvement. We are actively exploring additional strategies and methods to enhance our processes as we gather more insights.
We also welcome any suggestions or recommendations you may have on improving our KYC practices. Collaboration and feedback are invaluable as we work toward establishing a robust and efficient KYC framework.

**Comment by joshua-ne** [comment]
Further Introduction to the Dataset Card Initiative

**Rationale**
Through discussions with various investors, potential users of Fil-Plus project datasets, and network retrievers, it has become evident that a significant gap exists: accessing comprehensive and detailed information about datasets stored on the Filecoin network remains challenging. Specifically, stakeholders face difficulties in retrieving metadata, content identifiers (CIDs) required for data access, and information on the physical location of data—key factors that influence transfer speed and cost efficiency.

**Purpose and Expected Outcome**
The Dataset Card initiative aims to address these challenges by providing a standardized and comprehensive overview of open datasets. Each dataset card will include essential information, such as:
Content Description: A detailed overview of the dataset's structure and purpose.
Metadata: Version history, update frequency, licensing, and other relevant details.
Content Identifiers (CIDs): A complete list of CIDs associated with the dataset for seamless retrieval.
Indexing: Where applicable, a detailed index of the dataset's contents for easier navigation.
With a fully developed dataset card, users interested in retrieving datasets on the Filecoin network will have access to a clear, structured format containing all necessary information. This transparency will enable more efficient utilization of the network's resources.

**Impact**
If successfully implemented, this initiative will significantly enhance the usability of datasets on the Filecoin network. By making datasets more discoverable, accessible, and understandable, the Dataset Card initiative can drive increased adoption of Filecoin's decentralized storage solutions and promote greater data utilization.

**Current Progress and Challenges**
Despite its potential, developing and implementing the Dataset Card initiative presents some challenges. This initiative introduces an additional responsibility for clients and storage providers (SPs), potentially increasing their workload. To address this, we have initiated discussions with several clients and members of the Filecoin community to explore practical formats and tools that could streamline the process and ensure the cards are easy to create, maintain, and update.

**Conclusion**
The Dataset Card initiative represents a critical step toward improving the usability and transparency of datasets on the Filecoin network. By bridging existing gaps and providing essential information in a standardized format, this initiative has the potential to unlock significant value for investors, users, and network participants.

**Comment by joshua-ne** [comment]
Thank you, @galen-mcandrew, for your review. Later this week, we will open a new issue in this repo for the dataset card initiative for broader community discussions. 

And for the new retrieval tool, since it is triggered every per request, we will see how to incorporate to some of the existing tools. But for now, actually everyone can use this service by inputting the SP and client pair in this issue (https://github.com/joshua-ne/FIL_DC_Allocator_1022/issues/15), or by deploying their own bot using our open-source code and CID API.

**Comment by cryptoAmandaL** [comment]
@joshua-ne Can you provide deals which contains the initial samples of data presented by your client? 

@galen-mcandrew @filecoin-watchdog
